datafusion/row/src/reader.rs

```diff
@@ -142,7 +140,8 @@ impl<'a> RowReader<'a> {
         self.data = data;
     }
 
-    #[inline]
+    #[allow(dead_code)]
+    #[inline(always)]
     fn assert_index_valid(&self, idx: usize) {
```

Comment:
This still needed?

Reply:
This is not used in current code. I leave it here, this check can be invoked for each record batch.